lua/chatcommands/coinflip.lua

```diff
@@ -0,0 +1,153 @@
+	elseif not ply:PS_HasPoints(amount) then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh don't hae dooze points!")
+	elseif not target:PS_HasPoints(amount) then
+		-- Target doesn't have the required funds. Nothing happens, CoinFlipRequestTimeout will be called and timeout the roll.
+	end
+end
+
+function CoinFlipRemove(id)
+	timer.Destroy("CoinFlip" .. id)
+	table.remove(CoinFlips, id)
+end
+
+function checkCoinFlipRequest(toPlayer, points)
+	local coinflipFound = false
+	for fromID,j in pairs(CoinFlips) do
+		if j[1] == toPlayer:SteamID() and j[2] == points then
+			-- Coinflip Request Found
+			coinflipFound = true
+			finishCoinFlip(fromID, toPlayer)
+			break -- Only do the first request found.
+		end
+	end
+	if not coinflipFound then
+		ply:ChatPrint("[red]Yeh don't hae a coinflip request for tha amount!")
+		ply:ChatPrint("[orange]COINFLIPS:")
+		local index = 1
+		for fromID,j in pairs(CoinFlips) do
+			if j[1] == toPlayer:SteamID() then
+				local fromPlayer =  player.GetBySteamID(fromID)
+				local points = j[2]
+				if fromPlayer ~= nil then
+					ply:ChatPrint("[orange]" .. index .. ") [rainbow]" .. points .. "[fbc] from [gold]" .. fromPlayer:Nick())
+				end
+				index = index + 1
+			end
+		end
+	end
+end
+
+function hasCoinflipRequest(ply)
+	for fromID,data in pairs(CoinFlips) do
+		if data[1] == ply:SteamID() then
+			return true
+		end
+	end
+	return false
+end
+
+function finishCoinFlip(fromID, to)
+	-- Self explanatory, 'from' is the initiator player who's ID is the key in the table, 'to' is the other player.
+	local from = nil
+	local amount = CoinFlips[fromID][2]
+	-- Get Player from ID
+	for _,ply in ipairs( player.GetAll() ) do
+		if(ply:SteamID() == fromID) then from = ply end
+	end
+	if(from == nil) then
+		CoinFlipRemove(fromID) -- Remove request from CoinFlip because initiator left the server
+		to:ChatPrint("[red]The other bruv left, gotta cancel this coinflip!")
+	elseif from:PS_HasPoints(amount) and to:PS_HasPoints(amount) then -- Final Check, make sure they have funds still
+		CoinFlipRemove(fromID) -- Remove request because roll is starting
+		local heads = math.random() < 0.5 -- the "request from" player is always Heads.
+		BotSayGlobal("[edgy]" .. from:Nick() .. "[fbc] flipped a coin worth [rainbow]" .. (amount * 2) .. "[fbc] against [gold]".. to:Nick().. "[fbc] and...... [rainbow]" .. (heads and "Won" or "Lost") .."!")
+		from:ChatPrint("[fbc]You " .. (heads and "Won" or "Lost") .. " " .. amount .. ".")
+		to:ChatPrint("[fbc]You " .. (heads and "Won" or "Lost") .. " " .. amount .. ".")
+		-- Instead of taking the amount away from both and then giving the winner the amount x 2, simply remove/add here
+		if heads then
+			to:PS_TakePoints(amount)
+			from:PS_GivePoints(amount)
+		else
+			from:PS_TakePoints(amount)
+			to:PS_GivePoints(amount)
+		end
+	elseif not from:PS_HasPoints(amount) then
+		-- Initiator suddenly doesn't have the required funds. Nothing happens, show it off as an error.
```

Comment:
you don't need to worry about which of the two players doesn't have enough points, just throw the same error either way so the code is smaller